src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -451,13 +402,378 @@ def __init__(
+            stream_mode=stream_mode,
+            cache=cache,
+            stream_cache_path=stream_cache_path,
+        )
+
+        # If the electrical_series_location is not given, `find_electrical_series` will be called
+        # And returns a dictionary with the electrical_series_name as key and the location as value
+        # If there is only one electrical series, the electrical_series_name is set to the name of the series
+        if electrical_series_location is None:
+            available_electrical_series = _NWBHDF5RecordingExtractor.find_electrical_series(hdf5_file)
+            if electrical_series_name is None:
+                if len(available_electrical_series) == 1:
+                    electrical_series_name = list(available_electrical_series.keys())[0]
+                else:
+                    raise ValueError(
+                        "Multiple ElectricalSeries found in the file. "
+                        "Please specify the 'electrical_series_name' argument:"
+                        f"Available options are: {available_electrical_series}."
+                    )
+            else:
+                if electrical_series_name not in available_electrical_series:
+                    raise ValueError(
+                        f"'{electrical_series_name}' not found in the file. "
+                        f"Available options are: {available_electrical_series}"
+                    )
+            electrical_series_location = available_electrical_series[electrical_series_name]
+
+        self.electrical_series_location = electrical_series_location
+
+        # The indices in the electrode table corresponding to this electrical series
+        electrical_series = hdf5_file[electrical_series_location]
+        electrodes_indices = electrical_series["electrodes"][:]
+        # The table for all the electrodes in the nwbfile
+        electrodes_table = hdf5_file["/general/extracellular_ephys/electrodes"]
+        electrode_table_columns = electrodes_table.attrs["colnames"]
+
+        # Get sampling frequency
+        if "starting_time" in electrical_series.keys():
+            t_start = electrical_series["starting_time"][()]
+            sampling_frequency = electrical_series["starting_time"].attrs["rate"]
+        elif "timestamps" in electrical_series.keys():
+            t_start = electrical_series["timestamps"][0]
+            sampling_frequency = 1 / np.median(np.diff(electrical_series["timestamps"][:1000]))
+
+        # TimeSeries need to have either timestamps or rate
+        if sampling_frequency is None:
+            raise ValueError(
+                "Could not find rate information as both 'rate' and "
+                "'timestamps' are missing from the file. "
+                "Use the 'sampling_frequency' argument."
+            )
+
+        times_kwargs = dict(sampling_frequency=sampling_frequency, t_start=t_start)
+
+        # Extractors channel groups must be integers, but Nwb electrodes group_name can be strings
```

Comment:
Groups can also be strings. They are just any other property!

Reply:
Changed, it was carried out from main.

Reply:
Probably old ;)